### PR TITLE
Include all mode construction in Jenkins CI (Develop branch)

### LINKED ISSUE #604: [ENH] Allow speficying the build and library modes in the linux_build.py script
### Enhancement description

In order to build in different modes in the CI pipeline, we will need to specify the different build and libs modes for CMake.

### LINKED ISSUE #605: [FEAT] Remove the branch from the git submodule to avoid confussion
### Feature description

From now on, since we do merges back an forth between master and develop, the submodule of rticonnextdds-cmake-utils will be driven only by the commit, not the branch itself.

### LINKED ISSUE #606: [ENH] Build the 4 modes in develop CI
### Enhancement description

Enable building in the 4 combinations:
- `release/dynamic`
- `release/static`
- `debug/dynamic`
- `debug/static`

### PR DESCRIPTION
**Pull request by lulivi**
Fix #606 
Fix #605 
Fix #604 

### Summary

This PR adds the 4 build combination modes (Release/Debug, Static/Dynamic) to the Jenkins pipeline.

### Details and comments

Some files has been modified in order to be able to build in all the modes without problems.
- Please @pralhadsapre, can you check the routing and recorder examples?

### Checks

-   [x] I have updated the documentation accordingly.
-   [x] I have read the [CONTRIBUTING](https://github.com/rticommunity/rticonnextdds-examples/blob/develop/CONTRIBUTING.md) document.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -75,79 +166,77 @@ pipeline {
...
                 }
 
                 stage('Static Analysis') {
                     steps {
-                        publishChecks detailsURL: DETAILS_URL, name: STAGE_NAME,
-                            status: 'IN_PROGRESS', title: 'In progress', text: detailsText,
-                            summary: ':mag: Analysing all the examples...'
+                        publishPassedCheck(
+                            title: 'In progress',
+                            summary: ':mag: Analysing all the examples...',
```
Comment (rkorte) [inline Jenkinsfile:234]: Analyzing

(we should use American English spelling) ... but I'm not going to hold up the pull request for this

are these customer-facing messages?

**Comment by lulivi** [comment]
The output for all build modes but one was missing.

File: Jenkinsfile (hunk 1 above)
Comment (rkorte) [reply Jenkinsfile]: This still hasn't changed to "Analyzing" (with a z), but I'm not going to hold up the pull request for this

File: Jenkinsfile [hunk 2]
```diff
@@ -9,18 +9,107 @@
...
+    cmd += ' | tee $RTI_LOGS_FILE'
+    def returnCode = sh(
+        script: """#!/bin/bash
+            set -o pipefail
+            ${cmd}
+        """,
+        returnStatus: true,
+    )
+
+    sh 'python3 resources/ci_cd/jenkins_output.py'
```
Comment (fgaranda) [inline Jenkinsfile:80]: Do you need to add `${env.WORKSPACE}` here or it simply works.

File: Jenkinsfile [hunk 3]
```diff
@@ -75,100 +168,102 @@ pipeline {
                         sh 'tar zxvf connextdds-staging-${CONNEXTDDS_ARCH}.tgz unlicensed/'
 
                         sh '''
-                        cp ${RTI_INSTALLATION_PATH}/rti_connext_dds-*/lib/${CONNEXTDDS_ARCH}/openssl-1.*/* \
-                            ${RTI_INSTALLATION_PATH}/rti_connext_dds-*/lib/${CONNEXTDDS_ARCH}/
+                            cp ${RTI_INSTALLATION_PATH}/rti_connext_dds-*/lib/${CONNEXTDDS_ARCH}/openssl-1.*/* \
+                                ${RTI_INSTALLATION_PATH}/rti_connext_dds-*/lib/${CONNEXTDDS_ARCH}/
```
Comment (fgaranda) [inline Jenkinsfile:172]: Should we be doing openss-3*?

**Review by fgaranda — CHANGES_REQUESTED** [review]

**Review by ManuelJNunez — APPROVED** [review]

**Review by jpovedano — APPROVED** [review]

File: examples/recording_service/pluggable_storage/c++11/CMakeLists.txt [hunk 4]
```diff
@@ -19,15 +19,17 @@ if(NOT BUILD_SHARED_LIBS)
     set(msg
         "Static linking is not supported for this example. You can build it "
         "using the -DBUILD_SHARED_LIBS=ON CMake parameter")
-    message(FATAL_ERROR ${msg})
+    message(WARNING ${msg})
+    return()
```
Comment (pralhadsapre) [inline examples/recording_service/pluggable_storage/c++11/CMakeLists.txt:23]: Will the `return()` call stop the generate step? It wasn't super clear from the CMake documentation - https://cmake.org/cmake/help/latest/command/return.html

File: examples/recording_service/pluggable_storage/c++11/CMakeLists.txt [hunk 5]
```diff
@@ -19,15 +19,17 @@ if(NOT BUILD_SHARED_LIBS)
     set(msg
```
Comment (pralhadsapre) [inline examples/recording_service/pluggable_storage/c++11/CMakeLists.txt:19]: @lulivi for cases like these where a certain combination of the build isn't supported, it looks like we would just skip it. Is that what you intended?

Comment (lulivi) [reply examples/recording_service/pluggable_storage/c++11/CMakeLists.txt:19]: We want to skip them in order to build all the other examples. If this fails, no build would be done unless the correct build-link modes are selected for this example. It is intended.

**Review by pralhadsapre — APPROVED** [review]

**Review by fgaranda — APPROVED** [review]

**Pull request merged**